### PR TITLE
Add skyloong/Gk61 pro keyboard

### PR DESCRIPTION
**Pull request by JZ-Skyloong**
Compared to the V1 version, support for pluggable encoder modules has been added.

## Description

The key in the middle of the split space bar can be installed or replaced with a skyloong developed encoder module. Encoder module and key mounting sizes are fully compatible.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/skyloong/gk61/pro/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        KC_LSFT,     KC_Z,     KC_X,     KC_C,     KC_V,     KC_B,        KC_N,     KC_M,    KC_COMM,   KC_DOT,     KC_SLSH,               KC_RSFT,
+        KC_LCTL,  KC_LGUI,  KC_LALT,             KC_SPC,   KC_SPC,     KC_MUTE,               KC_SPC,  KC_RALT,      KC_APP,     KC_RCTL,               MO(1)
+    ),
+
+    [1] = LAYOUT_60_ansi(
+        KC_GRV,   KC_F1,    KC_F2,    KC_F3,    KC_F4,    KC_F5,       KC_F6,    KC_F7,    KC_F8,    KC_F9,    KC_F10,   KC_F11,   KC_F12,   _______,
+        _______,  RGB_TOG,  _______,  _______,  _______,  _______,     _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,
+        _______,  _______,  _______,  _______,  _______,  _______,     _______,  _______,  _______,  _______,  _______,  _______,            _______,
+        _______,  _______,  _______,  _______,  _______,  _______,     _______,  _______,  _______,  _______,  _______,            _______,
+        _______,  _______,  _______,                      _______,                                   _______,  _______,  _______,            _______
```
Comment (dunk2k) [inline keyboards/skyloong/gk61/pro/keymaps/default/keymap.c:33]:
```suggestion
    [1] = LAYOUT_all(
        KC_GRV,   KC_F1,    KC_F2,    KC_F3,    KC_F4,    KC_F5,       KC_F6,    KC_F7,    KC_F8,    KC_F9,    KC_F10,   KC_F11,   KC_F12,   _______,
        _______,  RGB_TOG,  _______,  _______,  _______,  _______,     _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,
        _______,  _______,  _______,  _______,  _______,  _______,     _______,  _______,  _______,  _______,  _______,  _______,            _______,
        _______,  _______,  _______,  _______,  _______,  _______,     _______,  _______,  _______,  _______,  _______,            _______,
        _______,  _______,  _______,             _______, _______,     _______,     _______,  _______,  _______,  _______,            _______
```

File: keyboards/skyloong/gk61/pro/info.json [hunk 2]
```diff
@@ -0,0 +1,366 @@
...
+        "console": false,
+        "encoder": true,
+        "encoder_map": true,
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": true,
+        "rgb_matrix": true,
+        "via": true
+    },
+    "keymaps": {},
```
Comment (dunk2k) [inline keyboards/skyloong/gk61/pro/info.json:31]:
```suggestion
```

File: keyboards/skyloong/gk61/pro/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    ),
+
+};
+
+#if defined(ENCODER_MAP_ENABLE)
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][2] = {
```
Comment (dunk2k) [inline keyboards/skyloong/gk61/pro/keymaps/default/keymap.c:39]:
```suggestion
const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
```

File: keyboards/skyloong/gk61/pro/info.json [hunk 4]
```diff
@@ -0,0 +1,366 @@
...
+        "encoder": true,
+        "encoder_map": true,
+        "rgb_matrix": true,
+        "via": true
+    },
+    "diode_direction": "ROW2COL",
+    "encoder": {
+        "enabled": true,
+        "rotary": [
+            {"pin_a": "C5", "pin_b": "C4"}
```
Comment (dunk2k) [inline keyboards/skyloong/gk61/pro/info.json:16]:
```suggestion
            {"pin_a": "C5", "pin_b": "C4", "resolution": 2}
```

File: keyboards/skyloong/gk61/pro/config.h [hunk 5]
```diff
@@ -0,0 +1,53 @@
+// Copyright 2023 linlin012 (@linlin012)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+/* Encoder Configuration */
+#ifdef ENCODER_ENABLE
+#    define ENCODER_RESOLUTION 2
```
Comment (dunk2k) [inline keyboards/skyloong/gk61/pro/config.h:8]: moved to `info.json`
```suggestion
```

File: keyboards/skyloong/gk61/pro/info.json [hunk 6]
```diff
@@ -0,0 +1,366 @@
...
+    "manufacturer": "skyloong",
+    "keyboard_name": "gk61 pro",
+    "maintainer": "linlin012",
+    "bootloader": "stm32duino",
+    "config_h_features": {
+        "encoder": true,
+        "encoder_map": true,
+        "rgb_matrix": true,
+        "via": true
+    },
```
Comment (dunk2k) [inline keyboards/skyloong/gk61/pro/info.json:11]:
```suggestion
```

File: keyboards/skyloong/gk61/pro/info.json [hunk 7]
```diff
@@ -0,0 +1,366 @@
...
+        "bootmagic": true,
+        "command": false,
+        "console": false,
+        "encoder": true,
+        "encoder_map": true,
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": true,
+        "rgb_matrix": true,
+        "via": true
```
Comment (dunk2k) [inline keyboards/skyloong/gk61/pro/info.json:29]:
```suggestion
```

File: keyboards/skyloong/gk61/pro/keymaps/via/keymap.c [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+        _______,  _______,  _______,  _______,  _______,  _______,     _______,  _______,  _______,  _______,  RGB_SPD,  RGB_SPI,            _______,
+        _______,  _______,  _______,  _______,  _______,  _______,     _______,  _______,  RGB_VAD,  RGB_VAI,  KC_UP,              _______,
+        _______,    TO(1),  _______,            _______,  _______,     _______,            _______,  KC_LEFT,  KC_DOWN,  KC_RIGHT,           _______
+    )
+};
+
+#if defined(ENCODER_MAP_ENABLE)
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][2] = {
+    [_WIN] = { ENCODER_CCW_CW(KC_VOLD, KC_VOLU) },
+    [_MAC] = { ENCODER_CCW_CW(KC_VOLD, KC_VOLU) },
```
Comment (dunk2k) [inline keyboards/skyloong/gk61/pro/keymaps/via/keymap.c:63]:
```suggestion
const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
    [_WIN] = { ENCODER_CCW_CW(KC_VOLD, KC_VOLU) },
    [_WIN_LOCK] = { ENCODER_CCW_CW(KC_TRNS, KC_TRNS) },
    [_MAC] = { ENCODER_CCW_CW(KC_TRNS, KC_TRNS) },
```

**Review by dunk2k — CHANGES_REQUESTED** [review]

**Comment by JZ-Skyloong** [comment]
There are too many comments when submitting PR, let me sort it out and resubmit it! Thank you for your suggestion. @dunk2k

**Pull request closed without merging**